source/extensions/clusters/aggregate/cluster_lb.cc

```diff
@@ -0,0 +1,23 @@
+#include "extensions/clusters/aggregate/cluster_lb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+Upstream::HostConstSharedPtr
+AggregateClusterLoadBalancer::chooseHost(Upstream::LoadBalancerContext* context) {
+  return chooseCluster()->loadBalancer().chooseHost(context);
+}
+
+Upstream::ThreadLocalCluster* AggregateClusterLoadBalancer::chooseCluster() const {
+  auto priority_pair = choosePriority(random_.random(), per_priority_load_.healthy_priority_load_,
+                                      per_priority_load_.degraded_priority_load_);
+
+  return priority_to_cluster_[priority_pair.first];
```

Comment:
does ignoring the second value here potentially mess with the load distribution when degraded endpoints are used?

i haven't done the math on this, but does the fact that whether we selected the priority because it fell in the healthy/degraded "bucket" is ignored when picking a host from the selected cluster impact the load distribution?

Reply:
>does ignoring the second value here potentially mess with the load distribution when degraded endpoints are used?
 >i haven't done the math on this, but does the fact that whether we selected the priority because it fell in the healthy/degraded "bucket" is ignored when picking a host from the selected cluster impact the load distribution?

Can you explain more about this?